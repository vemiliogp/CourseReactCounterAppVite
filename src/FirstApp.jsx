import PropTypes from "prop-types";

// const message = { name: "Victor", age: 28 };
// const getResult = (x, y) => x + y;

export const FirstApp = ({ title, subTitle, name }) => {
  // console.log(props);

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1>{getResult(5, 5)}</h1> */}
      {/* <code>{JSON.stringify(message)}</code> */}
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  name: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: "Victor",
  subTitle: "Not subtitle",
  // title: "Not title",
};
