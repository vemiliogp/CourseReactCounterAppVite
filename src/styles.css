html,
body {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  font-size: 1.3rem;
  margin-right: 15px;
  padding: 15px;
}
