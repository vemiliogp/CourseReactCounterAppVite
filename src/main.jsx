import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";

import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <CounterApp value={0} />
    {/* <FirstApp title="Hello, I am Satoru" /> */}
  </React.StrictMode>
);
