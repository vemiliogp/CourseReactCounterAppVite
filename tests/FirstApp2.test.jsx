import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Tests in <FirstApp />", () => {
  const title = "Hello, I'm Satoru";
  const subTitle = "I'm a subtitle";

  test("should match to snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show message 'Hello, I'm Satoru'", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug()
  });

  test("should show title in h1", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
    // screen.debug()
  });

  test("should show subtitle", () => {
    render(<FirstApp title={title} subTitle={subTitle} />); 

    expect(screen.getByText(subTitle)).toBeTruthy();
  })
});
