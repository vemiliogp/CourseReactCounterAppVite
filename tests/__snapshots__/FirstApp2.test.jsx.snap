// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match to snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, I'm Satoru
  </h1>
  <p>
    Not subtitle
  </p>
  <p>
    Victor
  </p>
</div>
`;
